drivers/nfs.py

```diff
@@ -282,11 +318,26 @@ def get_supported_nfs_versions(server):
                 cvi = ns[i].split()[1].split(",")
                 for j in range(len(cvi)):
                     cv.add(cvi[j])
-        return sorted(cv & valid_versions)
-    except:
-        util.SMlog("Unable to obtain list of valid nfs versions")
+        res = sorted(cv & valid_versions)
+    except Exception:
+        util.SMlog("Unable to obtain list of valid nfs versions with rpcinfo, trying NSFv4")
+
+    try:
+        mountpoint = "%s/%s" % NFS4_TMP_MOUNTPOINT, server
+        soft_mount(mountpoint, server, NSFv4_PSEUDOFS, transport, nfsversion='4')
+        util.pread2([NFS_STAT, '-m'])
+        unmount(mountpoint, NSFv4_PSEUDOFS)
+        if res is None:
+            res = ['4']
+        else:
+            res.add('4')
+    except Exception:
+        util.SMlog("Unable to obtain list of valid nfs versions with NSFv4 pseudo FS mount")
```

Comment:
Is this needed at all when res is not None? IIRC, when the server supports both v3 and v4, rpcinfo already lists 4 among the versions, so there's no need to try to mount the pseudo filesystem from NFSv4 in this situation.